pkg/validation/limits.go

```diff
@@ -1117,24 +1117,35 @@ func (o *Overrides) BlockIngestionStatusCode(userID string) int {
 	return o.getOverridesForUser(userID).BlockIngestionStatusCode
 }
 
+// BlockIngestionPolicyUntil returns the time until the ingestion policy is blocked for a given user.
+// Order of priority is: named policy block > global policy block. The global policy block is enforced
+// only if the policy is empty.
 func (o *Overrides) BlockIngestionPolicyUntil(userID string, policy string) time.Time {
 	limits := o.getOverridesForUser(userID)
-	if limits == nil || limits.BlockIngestionPolicyUntil == nil {
-		return time.Time{} // Zero time means no blocking
+
+	if forPolicy, ok := limits.BlockIngestionPolicyUntil[policy]; ok {
+		return time.Time(forPolicy)
 	}
 
-	if blockUntil, ok := limits.BlockIngestionPolicyUntil[policy]; ok {
-		return time.Time(blockUntil)
+	// We enforce the global policy on streams not matching any policy
+	if policy == "" {
+		if forPolicy, ok := limits.BlockIngestionPolicyUntil[GlobalPolicy]; ok {
+			return time.Time(forPolicy)
+		}
 	}
+
 	return time.Time{} // Zero time means no blocking
 }
 
 func (o *Overrides) EnforcedLabels(userID string) []string {
 	return o.getOverridesForUser(userID).EnforcedLabels
 }
 
+// PolicyEnforcedLabels returns the labels enforced by the policy for a given user.
+// The output is the union of the global and policy specific labels.
 func (o *Overrides) PolicyEnforcedLabels(userID string, policy string) []string {
-	return o.getOverridesForUser(userID).PolicyEnforcedLabels[policy]
+	limits := o.getOverridesForUser(userID)
+	return append(limits.PolicyEnforcedLabels[GlobalPolicy], limits.PolicyEnforcedLabels[policy]...)
```

Comment:
if there's no enforced labels for the global policy this would append an empty slice (i think?), isn't it a problem?

Reply:
yes, but appending an empty slice yields the same slice. If none are set, this would be an empty slice.